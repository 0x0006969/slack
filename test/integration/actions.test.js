const request = require('supertest');

const { probot, models } = require('.');
const fixtures = require('../fixtures');

describe('Actions', async () => {
  beforeEach(async () => {
    const { SlackWorkspace, SlackUser } = models;
    const workspace = await SlackWorkspace.create({
      slackId: 'T000A',
      accessToken: 'xoxa-token',
    });
    await SlackUser.create({
      slackId: 'U88HS',
      slackWorkspaceId: workspace.id,
    });
  });
  test('When an action\'s name is "cancel", then it is immediately deleted', async () => {
    await request(probot.server).post('/slack/actions').send({
      payload: JSON.stringify(fixtures.slack.action.cancel()),
    })
      .expect(200)
      .expect((res) => {
        expect(res.body).toMatchSnapshot();
      });
  });
});
