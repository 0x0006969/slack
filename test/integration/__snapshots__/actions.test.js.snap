// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions An unknown callback_id returns a 500 1`] = `Set {}`;

exports[`Actions An unknown callback_id returns a 500 2`] = `Map {}`;

exports[`Actions When an action's name is "cancel", then it is immediately deleted 1`] = `
Object {
  "delete_original": true,
}
`;

exports[`Actions When an action's name is "cancel", then it is immediately deleted 2`] = `Set {}`;

exports[`Actions When an action's name is "cancel", then it is immediately deleted 3`] = `Map {}`;

exports[`Actions responds with 400 when POSTed to directly invalid format 1`] = `Set {}`;

exports[`Actions responds with 400 when POSTed to directly invalid format 2`] = `Map {}`;

exports[`Actions responds with 400 when POSTed to directly invalid verification token 1`] = `Set {}`;

exports[`Actions responds with 400 when POSTed to directly invalid verification token 2`] = `Map {}`;
